routes/getUserInfo.js

```diff
@@ -0,0 +1,38 @@
+var express = require("express");
+var Q = require('q');
+var router = express.Router();
+
+impredis = require("../imp_services/impredis.js");
+
+router.route('/').get(function(req, res) {
+    var implogging = require('../imp_services/implogging');
+    var db = require("../imp_services/impdb.js").connect();
+    impredis.get(req.cookies.IMPId, function (val, error) {
+        if (error) {
+            res.send("error: " + error);
+        }
+        else {
+            Q.fcall(db.beginTransaction())
+                .then(db.query("USE " + db.databaseName))
+                .then(db.query("CALL GetUserByUsername ('"+val.username+"')"))
+                .then(function(rows, columns){
+                    var response = rows[0][0][0];
+                    res.send({
+                        FirstName: response.FirstName,
```

Comment:
I would've thought you'd have to do a parse here before accessing the response fields. apparently not though.